app/models/course/discussion/topic.rb

```diff
@@ -22,7 +22,30 @@ def self.global_topic_models
       where(actable_type: global_topic_models.map(&:name))
   end)
 
-  scope :ordered_by_updated_at, -> { order(updated_at: :desc) }
+  # Returns the topics from the user(s) specified.
+  # TODO: Remove the workaround and use the following implementation which is more cleaner after
+  # https://github.com/activerecord-hackery/squeel/issues/344 is closed, a similar rails issue is
+  # https://github.com/rails/rails/issues/20077.
+  #   scope :from_user, (lambda do |user_id|
+  #     where do
+  #       id >> Course::Assessment::Answer::ProgrammingFileAnnotation.from_user(user_id) |
+  #         id >> Course::Assessment::Answer.from_user(user_id)
+  #     end
+  #   end)
+  #
+  # @param[Integer|Array<Integer>] user_id, the id(s) of the user(s).
+  # @return[Array<Course::Discussion::Topic>]
+  scope :from_user, (lambda do |user_id|
+    where(<<-SQL)
+      course_discussion_topics.id IN (#{Course::Assessment::Answer::ProgrammingFileAnnotation.from_user(user_id).to_sql})
+      OR
+      course_discussion_topics.id IN (#{Course::Assessment::Answer.from_user(user_id).to_sql})
+    SQL
+  end)
+
+  scope :ordered_by_updated_at, ->() { order(updated_at: :desc) }
```

Comment:
Don't think there is a need for the `()`? `scope :ordered_by_updated_at, -> { order(updated_at: :desc) }